pkg/controller/fakeclusterinstall/fakeclusterinstall_controller.go

```diff
@@ -0,0 +1,313 @@
+		hivev1.ClusterInstallFailed,
+		hivev1.ClusterInstallStopped,
+		hivev1.ClusterInstallRequirementsMet,
+	}
+
+	var anyChanged bool
+	for _, condType := range conditionTypes {
+		c := controllerutils.FindClusterInstallCondition(fci.Status.Conditions, condType)
+		if c == nil {
+			logger.WithField("condition", condType).Info("initializing condition with Unknown status")
+			newConditions, changed := controllerutils.SetClusterInstallConditionWithChangeCheck(
+				fci.Status.Conditions,
+				condType,
+				corev1.ConditionUnknown,
+				"",
+				"",
+				controllerutils.UpdateConditionAlways)
+			fci.Status.Conditions = newConditions
+			anyChanged = anyChanged || changed
+		}
+	}
+
+	if anyChanged {
+		err := updateClusterInstallStatus(r.Client, fci, logger)
+		return reconcile.Result{}, err
+	}
+
+	// Check if we're already Stopped=True and return if so.
+	// If not, we should be Stopped=False as we're actively working.
+	stoppedCond := controllerutils.FindClusterInstallCondition(fci.Status.Conditions, hivev1.ClusterInstallStopped)
+	if stoppedCond == nil {
+		errMsg := fmt.Sprintf("%s condition is nil but should be set", hivev1.ClusterInstallStopped)
+		return reconcile.Result{}, errors.New(errMsg)
+	}
+	if stoppedCond.Status == corev1.ConditionTrue {
+		logger.Info("Stopped=True no processing necessary")
+		return reconcile.Result{}, nil
+	} else if stoppedCond.Status == corev1.ConditionUnknown {
+		newConditions, changed := controllerutils.SetClusterInstallConditionWithChangeCheck(
+			fci.Status.Conditions,
+			hivev1.ClusterInstallStopped,
+			corev1.ConditionFalse,
+			"InProgress",
+			"Cluster install in progress",
+			controllerutils.UpdateConditionIfReasonOrMessageChange)
+		if changed {
+			fci.Status.Conditions = newConditions
+			err := updateClusterInstallStatus(r.Client, fci, logger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Fetch corresponding ClusterDeployment instance
+	cd := &hivev1.ClusterDeployment{}
+	switch err = r.Get(context.TODO(), request.NamespacedName, cd); {
+	case apierrors.IsNotFound(err):
+		// TODO: assuming same name, add explicit reference of some kind between cluster install and cluster deplopyment
+		logger.WithField("clusterDeployment", request.NamespacedName).Info("ClusterDeployment not found")
+		return reconcile.Result{}, nil
+	case err != nil:
+		logger.WithError(err).Error("Error getting ClusterDeployment")
+		return reconcile.Result{}, err
+	}
+	if !cd.DeletionTimestamp.IsZero() {
+		logger.Debug("ClusterDeployment has been deleted")
+		return reconcile.Result{}, nil
+	}
+
+	// Simulate 30 second wait for RequirementsMet condition to go True:
+	reqsCond := controllerutils.FindClusterInstallCondition(fci.Status.Conditions, hivev1.ClusterInstallRequirementsMet)
+	if reqsCond == nil {
+		errMsg := fmt.Sprintf("%s condition is nil but should be set", hivev1.ClusterInstallRequirementsMet)
+		return reconcile.Result{}, errors.New(errMsg)
+	}
+	if reqsCond.Status == corev1.ConditionUnknown {
```

Comment:
maybe do a switch here, that would read cleaner..?

Reply:
Sounds good.